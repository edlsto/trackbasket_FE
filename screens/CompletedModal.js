import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MaterialIcons } from 'react-native-vector-icons';
import { Button } from '../components/Button';

const CompletedModal = ({ navigation: { goBack } }) => {
  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <View>
          <MaterialIcons name="done" color="green" size={120} />
        </View>
        <Text style={styles.introtext}>
          You have shopped all items on this list!
        </Text>
        <Button
          text="CLOSE"
          customStyles={{ backgroundColor: '#59DE7E', marginTop: 20 }}
          onPress={() => goBack()}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  innerContainer: {
    width: 300,
    marginBottom: 50,
    alignItems: 'center',
  },
  introtext: {
    fontSize: 30,
    marginBottom: 20,
    textAlign: 'center',
  },
});

export default CompletedModal;
