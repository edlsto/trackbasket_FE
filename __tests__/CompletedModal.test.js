import React from 'react';
import CompletedModal from '../screens/CompletedModal';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer
    .create(<CompletedModal navigation={{ goBack: jest.fn() }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
