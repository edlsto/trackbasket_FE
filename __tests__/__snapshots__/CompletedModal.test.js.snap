// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 50,
        "width": 300,
      }
    }
  >
    <View>
      <Text />
    </View>
    <Text
      style={
        Object {
          "fontSize": 30,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      You have shopped all items on this list!
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#59DE7E",
          "marginTop": 20,
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 10,
          "width": 220,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "HelveticaNeue-Bold",
              "fontSize": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        CLOSE
      </Text>
    </View>
  </View>
</View>
`;
